File: app/src/main/java/kr/djspi/pipe01/Const.java
package kr.djspi.pipe01;

import kr.djspi.pipe01.dto.PipeType.PipeTypeEnum;

public final class Const {

    public static final String URL_SPI = "http://35.187.193.145/";
    static final String URL_TEST = "http://192.168.0.33/";

    public static final String TAG_PIPE = "pipe";
    public static final String TAG_SHAPE = "shape";
    public static final String TAG_SUPERVISE = "supervise";
    public static final String TAG_POSITION = "position";
    public static final String TAG_DIRECTION = "direction";
    public static final String TAG_DISTANCE = "distance";
    public static final String TAG_LOCATION = "location";
    public static final String TAG_SURVEY = "survey";
    public static final String TAG_PREVIEW = "preview";
    public static final String TAG_PHOTO = "photo";
    public static final String[] PIPE_SHAPES = {"직진형", "T분기형", "엘보형", "관말형"};
    public static final String[] PIPE_DIRECTIONS = {null, "", "out", "", "outl", null, "outr", "", "in", ""};
    public static final PipeTypeEnum[] PIPE_TYPE_ENUMS = PipeTypeEnum.values();

    static final int REQUEST_MAP = 30001;
    static final int REQUEST_CAPTURE_IMAGE = 10001;
    static final int REQUEST_GALLERY = 10002;

    public static final int RESULT_PASS = 200;
    public static final int RESULT_FAIL = 400;
}

File: app/src/main/java/kr/djspi/pipe01/Const.kt
package kr.djspi.pipe01

import kr.djspi.pipe01.dto.PipeType.PipeTypeEnum

object Const {
//    const val TAG_PIPE = "pipe"
    const val TAG_SHAPE = "shape"
//    const val TAG_SUPERVISE = "supervise"
    const val TAG_POSITION = "position"
    const val TAG_DIRECTION = "direction"
    const val TAG_DIRECTION_ELB135 = "direction2"
    const val TAG_DISTANCE = "distance"
    const val TAG_LOCATION = "location"
    const val TAG_SURVEY_SPI = "survey"
    const val TAG_SURVEY_PIPE = "survey2"
    const val TAG_SURVEY = "survey3"
    const val TAG_PREVIEW = "preview"
    const val TAG_PHOTO = "photo"
    val PIPE_DIRECTIONS = arrayOf("", "", "out", "", "outl", "", "outr", "", "in")
    val PIPE_DIRECTIONS_ELB135 = arrayOf("", "snw", "sne", "nsw", "nse", "esw", "enw", "wne", "wse")
    val PIPE_TYPE_ENUMS = PipeTypeEnum.values()

    internal const val REQUEST_MAP = 30001
    internal const val REQUEST_CAPTURE_IMAGE = 10001
    internal const val REQUEST_GALLERY = 10002
    internal const val REQUEST_APP_UPDATE = 40001

    const val RESULT_PASS = 200
    const val RESULT_FAIL = 400
}
